modules/sor/utils/data.ts

```diff
@@ -40,182 +82,109 @@ export async function getBasePoolsFromDb(
-                swapEnabled: true,
-                isPaused: false,
-            },
-        },
-        include: {
-            dynamicData: true,
-            tokens: { include: { token: true } },
-        },
-    });
-
-    return pools;
-}
-
-async function getFilteredPools(
-    chain: Chain,
-    protocolVersion: number,
-    considerPoolsWithHooks: boolean,
-    type: { in: PrismaPoolType[] },
-): Promise<PrismaPoolAndHookWithDynamic[]> {
-    const poolIdsToExclude = config[chain].sor?.poolIdsToExclude ?? [];
-
-    const [pools, lbps] = await Promise.all([
-        getPrimaryPools(chain, protocolVersion, type, poolIdsToExclude),
-        getLiquidityBootstrappingPools(chain, protocolVersion, poolIdsToExclude),
+    const [pools, dynamicData, poolTokens, tokens] = await Promise.all([
+        prisma.prismaPool
+            .findMany({
+                where: { chain, type, id: { notIn: [...poolsToIgnore] } },
+            })
+            .then((records) => Object.fromEntries(records.map((record) => [record.id, record]))),
+        prisma.prismaPoolDynamicData
+            .findMany({
+                // TODO: Narrow down the select, but need to change the return type to SOR internal one first
+                // select: { id: true, isPaused: true, swapEnabled: true },
+                where: {
+                    chain,
+                    totalSharesNum: { gt: 0.000000000001 },
+                    swapEnabled: true,
+                    isPaused: false,
+                    id: { notIn: [...poolsToIgnore] },
+                    OR: [
+                        {
+                            totalLiquidity: { gte: 100 },
+                        },
+                        {
+                            chain: 'SEPOLIA',
+                        },
+                        {
+                            pool: {
+                                type: 'LIQUIDITY_BOOTSTRAPPING',
+                            },
+                        },
+                    ],
+                },
+            })
+            .then((records) => Object.fromEntries(records.map((record) => [record.id, record]))),
+        prisma.prismaPoolToken
+            .findMany({
+                where: {
+                    pool: {
+                        chain,
+                        type,
+                        id: { notIn: [...poolsToIgnore] },
+                    },
+                },
+            })
+            .then((records) => _.groupBy(records, 'poolId') as Record<string, typeof records>),
+        tokenService
+            .getTokens(chain)
+            .then((tokens) => Object.fromEntries(tokens.map((token) => [token.address, token]))),
     ]);
 
-    const filteredPools = [...filterPoolsByHooks(pools, considerPoolsWithHooks), ...lbps];
-
-    return filteredPools;
+    const setWithDynamicDataIds = new Set(Object.keys(dynamicData));
+    const intersection = [...new Set(Object.keys(pools))].filter((id) => setWithDynamicDataIds.has(id));
```

Comment:
Is this intersection check actually needed? I mean, isn't dynamic data poolIds always a subset of poolIds?

Reply:
should be, but I've seen it happening

Reply:
Actually the other way around, pool didn't have associated dynamic part – perhaps it's not needed, but I assumed it's better to be on the safe side and do the interpolation.